### ISSUE TITLE
`keys` missing for SimpleVector

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#26679]
Many packages do reflection on fieldnames and end up doing something like:
```
julia> findfirst(==(:foo), Core.svec(:foo, :bar))
ERROR: MethodError: no method matching keys(::Core.SimpleVector)
Closest candidates are:
  keys(::Cmd) at process.jl:838
  keys(::Tuple) at tuple.jl:42
  keys(::Tuple, ::Tuple...) at tuple.jl:48
  ...
Stacktrace:
 [1] pairs(::Core.SimpleVector) at ./abstractdict.jl:138
 [2] findfirst(::Base.Fix2{typeof(==),Symbol}, ::Core.SimpleVector) at ./array.jl:1682
 [3] top-level scope
```
Needs an implementation of:
```
keys(s::Core.SimpleVector) = 1:length(s)
```
and a test.

**Comment by Ramit-Pahwa** [comment]
Hi, would like to work on this issue,  can you direct me how to start, I am new to julia.

**Comment by Ramit-Pahwa** [comment]
@Keno would like to work on this, can you suggest me a starting point ?

**Comment by JeffBezanson** [comment]
Methods for `SimpleVector` are in `base/essentials.jl`. The suggested definition can be added there.

**Comment by JeffBezanson** [comment]
Though should probably return `OneTo(length(s))` instead of `1:length(s)`.

**Comment by Ramit-Pahwa** [comment]
#26711

**Comment by Ramit-Pahwa** [comment]
Have sent a PR with fix . #26711 @JeffBezanson ,How would i go about writing the test.

**Comment by Ramit-Pahwa** [comment]
https://github.com/JuliaLang/julia/pull/26711/commits/197cfe3b3c225acf2e95671c19b464d4f4333078

**Comment by JeffBezanson** [comment]
Fixed by #26711.
